cmd/userdocker/main.go

```diff
@@ -191,73 +167,23 @@ func startDocker(cfg *config.CloudConfig) (string, types.HijackedResponse, error
 
 	if dockerCfg.TLS {
 		if err := writeCerts(cfg); err != nil {
-			return "", types.HijackedResponse{}, err
+			return err
 		}
 	}
 
-	cmd := []string{"env"}
+	info, err := client.ContainerInspect(context.Background(), storageContext)
+	if err != nil {
+		return err
+	}
+
+	cmd := []string{"docker-runc", "exec", "--", info.ID, "env"}
 	log.Info(dockerCfg.AppendEnv())
 	cmd = append(cmd, dockerCfg.AppendEnv()...)
 	cmd = append(cmd, DOCKER_COMMAND)
 	cmd = append(cmd, args...)
 	log.Infof("Running %v", cmd)
 
-	resp, err := client.ContainerExecCreate(context.Background(), types.ExecConfig{
-		Container:    storageContext,
-		Privileged:   true,
-		AttachStderr: true,
-		AttachStdout: true,
-		Cmd:          cmd,
-	})
-	if err != nil {
-		return "", types.HijackedResponse{}, err
-	}
-
-	attachResp, err := client.ContainerExecAttach(context.Background(), resp.ID, types.ExecConfig{
-		Detach:       false,
-		AttachStderr: true,
-		AttachStdout: true,
-	})
-	if err != nil {
-		return "", types.HijackedResponse{}, err
-	}
-
-	return resp.ID, attachResp, nil
-}
-
-func getDockerProcess() (*os.Process, error) {
-	pidBytes, err := waitForFile(DOCKER_PID_FILE)
-	if err != nil {
-		return nil, err
-	}
-	dockerPid, err := strconv.Atoi(string(pidBytes))
-	if err != nil {
-		return nil, err
-	}
-	return os.FindProcess(dockerPid)
-}
-
-func handleTerm(process *os.Process) {
-	term := make(chan os.Signal)
-	signal.Notify(term, syscall.SIGTERM)
-	go func() {
-		<-term
-		process.Signal(syscall.SIGTERM)
-	}()
-}
-
-func waitForFile(file string) ([]byte, error) {
-	for {
-		contents, err := ioutil.ReadFile(file)
-		if os.IsNotExist(err) {
-			log.Infof("Waiting for %s", file)
-			time.Sleep(1 * time.Second)
-		} else if err != nil {
-			return nil, err
-		} else {
-			return contents, nil
-		}
-	}
+	return syscall.Exec("/usr/bin/docker-runc", cmd, os.Environ())
```

Comment:
If you are doing `syscall.Exec` then you don't need to symlink.  Just change to `syscall.Exec("/usr/bin/ros", cmd, os.Environ())`